doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,254 @@
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "false"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "100"
+                }
+            ]
+        },
+        {
+            "name": "any FAN absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "false"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "100"
+                }
+            ]
+        },
+        {
+            "name": "all FAN and PSU presence",
+            "conditions": [
+                {
+                    "type": "fan.all.presence"
+                },
+                {
+                    "type": "psu.all.presence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "thermal_control.control",
+                    "status": "true"
+                },
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "65"
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### Invalid format policy file
+
+In invalid_format_policy.json, the file content is not valid JSON at all. A file with content "invalid" should be good for test purpose.
+
+#### Invalid value policy file
+
+In invalid_value_policy.json, the file content contains minus value of target speed. We couldn't cover all invalid value here because there are too many possibilities. The major purpose of this configuration file is to verify thermal control daemon would not crash while loading such an invalid configuration files. For other negative test cases, they are already covered by unit test. The invalid_value_policy.json content is like:
+
+```json
+{
+    "info_types": [
+        {
+            "type": "fan_info"
+        },
+        {
+            "type": "psu_info"
+        }
+    ],
+    "policies": [
+        {
+            "name": "any PSU absence",
+            "conditions": [
+                {
+                    "type": "fan.any.absence"
+                }
+            ],
+            "actions": [
+                {
+                    "type": "fan.all.set_speed",
+                    "speed": "-1"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Test Cases
+
+### Show FAN Status Test
+
+Show FAN status test verifies that all FAN related information can be shown correctly via `show platform fanstatus`.
+
+#### Procedure
+
+1. Testbed setup.
+2. Unlink FAN related sysfs and fill mock data for "presence", "speed", "status", "target_speed", "led status".
```

Comment:
What do you mean by "unlink"? There is no standard sysfs path for such data in SONiC I think. Another platform could read it from registers directly and may not have sysfs entries.

Reply:
Yes, "unlink" and most "mock" would only work for mellanox platform. As I can't mock data for other vendors, I will use code like " if dut.facts["asic_type"] in ["mellanox"]:" to make sure these test cases would not affect other vendor.

Reply:
I would suggest not to use if mellanox conditions.
Instead, I suggest to define an interface to mock data from platform.
Each vendor will implement its own way to do that.
Mellanox will implement it by unlikning the sysfs file.
If implementation of  mocking is not available skip the test

Reply:
I will adjust my test code.